user_tools/src/spark_rapids_tools/tools/qualification_stats_report.py

```diff
@@ -0,0 +1,117 @@
+from logging import Logger
+
+import pandas as pd
+import fire
+
+from spark_rapids_pytools.common.utilities import ToolLogging
+
+
+@dataclass
+class SparkQualificationStats:
+    """
+    Encapsulates the logic to generate the Qualification Stats Report.
+    """
+    logger: Logger = field(default=None, init=False)
+    unsupported_operators_file: str
+    stages_file: str
+    output_file: str
+    unsupported_operators_df: pd.DataFrame = field(default=None, init=False)
+    stages_df: pd.DataFrame = field(default=None, init=False)
+    result_df: pd.DataFrame = field(default=None, init=False)
+
+    def __init__(self, unsupported_operators_file: str, stages_file: str, output_file: str):
+        self.unsupported_operators_file = unsupported_operators_file
+        self.stages_file = stages_file
+        self.output_file = output_file
+        self.logger = ToolLogging.get_and_setup_logger('rapids.tools.qualification.stats')
+
+    def load_data(self):
+        try:
+            self.logger.info('Loading data from CSV files...')
+            self.unsupported_operators_df = pd.read_csv(self.unsupported_operators_file)
+            self.stages_df = pd.read_csv(self.stages_file)
+
+            # Convert durations from milliseconds to seconds
+            self.unsupported_operators_df['Stage Duration'] = (
+                    self.unsupported_operators_df['Stage Duration'] / 1000)
+            self.unsupported_operators_df['App Duration'] = (
+                    self.unsupported_operators_df['App Duration'] / 1000)
+            self.stages_df['Stage Task Duration'] = self.stages_df['Stage Task Duration'] / 1000
+            self.stages_df['Unsupported Task Duration'] = (
+                    self.stages_df['Unsupported Task Duration'] / 1000)
+        except Exception as e:  # pylint: disable=broad-except
+            self.logger.error('Error loading data: %s', e)
+
+    def merge_dataframes(self):
+        try:
+            self.logger.info('Merging dataframes...')
+            # Merge unsupported_operators_df with stages_df on App ID and Stage ID
+            merged_df = pd.merge(self.unsupported_operators_df, self.stages_df,
+                                 on=['App ID', 'Stage ID'])
+
+            agg_unsupported_df = (merged_df.groupby(['App ID', 'SQL ID', 'Unsupported Operator']).agg(
+                Count=('Unsupported Operator', 'size'),
+                Impacted_Stage_Duration=('Stage Duration', 'sum'),
+                App_Duration=('App Duration', 'sum'),
+                Stage_Task_Duration=('Stage Task Duration', 'sum')
+            ).reset_index())
+
+            agg_unsupported_df['% of Stage Duration'] = (
+                    (agg_unsupported_df['Impacted_Stage_Duration'] /
+                     agg_unsupported_df['App_Duration']) * 100)
+
+            agg_unsupported_df['Supported'] = False
+            final_df = agg_unsupported_df.rename(columns={
+                'App ID': 'AppId',
```

Comment:
These seems inconsistent. 
1. We rename some columns to add space -`'Impacted_Stage_Duration': 'Impacted Stage duration'`
2. We rename some columns to remove space `'SQL ID': 'SQLID'`

Similarly,
1. We rename some columns to remove underscore - `'Impacted_Stage_Duration': 'Impacted Stage duration'` 
2. We rename some columns to add underscore - `'Unsupported Operator': 'Operator_Name'`